services/libs/aster-std/src/syscall/stat.rs

```diff
@@ -53,12 +53,18 @@ pub fn sys_fstatat(
         "dirfd = {}, filename = {:?}, stat_buf_ptr = 0x{:x}, flags = {:?}",
         dirfd, filename, stat_buf_ptr, flags
     );
+
+    if filename.is_empty() {
+        if !flags.contains(StatFlags::AT_EMPTY_PATH) {
```

Comment:
> flags can either be 0, or include one or more of the following flags ORed:
>
> AT_NO_AUTOMOUNT (since Linux 2.6.38)
Don't automount the terminal ("basename") component of pathname if it is a directory that is an automount point. > This allows the caller to gather attributes of an automount point (rather than the location it would mount). This flag can be used in tools that scan directories to prevent mass-automounting of a directory of automount points. The AT_NO_AUTOMOUNT flag has no effect if the mount point has already been mounted over.
>
> AT_SYMLINK_NOFOLLOW
> If pathname is a symbolic link, do not dereference it: instead return information about the link itself, like [lstat](https://linux.die.net/man/2/lstat)(2). (By default, fstatat() dereferences symbolic links, like [stat](https://linux.die.net/man/2/stat)(2).)

The man page of [fstatat](https://linux.die.net/man/2/fstatat) does not mention this flag.

Reply:
The manual in https://man7.org/linux/man-pages/man2/lstat.2.html mentions the flag. After I checked the code of Linux, it uses the flag.
```
	if (flags & ~(AT_SYMLINK_NOFOLLOW | AT_NO_AUTOMOUNT | AT_EMPTY_PATH |
		      AT_STATX_SYNC_TYPE))
		return -EINVAL;
```

Reply:
Thanks for the explanation!